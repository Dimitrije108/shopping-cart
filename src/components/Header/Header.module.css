.headerCont {
	background-color: rgb(0, 0, 0);
	border-bottom: 1px solid rgb(214, 214, 214);
}

.stickyHeaderCont {
	background-color: rgba(0, 0, 0, 0.5);
	border-bottom: 1px solid rgb(214, 214, 214);

	position: sticky;
	top: 0px;

	z-index: 3;
}

.headerCont.transparentHeader {
	background-color: rgba(0, 0, 0, 0.5);

	position: relative;
  z-index: 2;
}

.headerCont.contactHeader {
	background-color: rgba(0, 0, 0, 0.5);

	background-image: url("/src/assets/images/galaxy_map2.jpg");
	background-size: cover;
	background-repeat: no-repeat;
	background-position: 50% 40%;
  background-blend-mode: overlay;
}

.logoCont{
	padding: 1.5rem 1rem 0 1rem;

	display: grid;
	grid-template-columns: 1fr 2fr 1fr;
	justify-content: center;
	align-items: end;
}

.githubLink {
	align-self: start;
}

.githubLink.sticky {
	margin: 0 1rem 0 3rem;
}

.githubLink.sticky > img  {
	width: 28px;
}

.githubLink > img {
	width: 30px;
}

.logoHeading {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	
	position: relative;
}

.logoLink {
	padding-bottom: 3.3rem;
}

.logoLink > img {
	width: 150px;
}

.shipyardLink {
	position: absolute;
	top: 50px;
}

.shipyardLogo {
	font-family: Imperial;
	font-size: 3.5rem;
	color: white;
}

.navbarCont {
	margin: 0 auto;
}

.navbarCont.sticky {
	padding-top: 1rem;

	background-color: rgba(0, 0, 0, 0.5);

	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
}

.navbar {
	font-size: 1.2rem;

	display: flex;
	justify-content: center;
}

.navbar > li {
	margin: 0 2rem;
	padding: 0 2rem 6px 2rem;

	transition: all 0.3s ease;
}

.navbar > li:hover {
	color: white;
}

.navbar.sticky > li {
	padding: 0 2rem 0 2rem;
}

.starshipsNav {
	position: relative;
	z-index: 3;
}

.homeNav.active,
.starshipsNav.active, 
.contactNav.active {
	color: white;
	border-bottom: 1px solid white;
}

.dropdown {
	padding: 1rem 25px 1rem 26px;

	color: rgb(221, 221, 221);
	background-color: rgba(0, 0, 0, 0.7);
	border-left: 1px solid rgb(214, 214, 214);
	border-right: 1px solid rgb(214, 214, 214);
	border-bottom: 1px solid rgb(214, 214, 214);

	position: absolute;
	top: 37px;
	left: 0;

	z-index: 2;

	visibility: hidden;

	animation: none;

	opacity: 0;
  overflow: hidden;
  transition: opacity 0.5s ease-in-out;
}

.dropdown.sticky {
	top: 43.5px;
}

.dropdown.visible {
	visibility: visible;

  opacity: 1;
  animation: growDropdown 1s ease-in-out;
}

.cart {
	top: 36px;
}

.dropdown > li {
	padding: 6px 0;
}

.dropdown > li:hover {
	color: white;
}

.cartIconLink {
	justify-self: end;
	position: relative;
}

.cartIconLink.sticky {
	margin: 0 3rem 0 1rem;
}

.cartIconLink.sticky > img {
	width: 35px;
}

.cartIconLink > img {
	width: 55px;
}

.cartQuantity {
	width: 26px;
  height: 26px;
  
	color: #fff;
  font-size: 1rem;
	background-color: rgb(214, 0, 0);
  text-align: center;

	border: 2px solid #000;
  border-radius: 50%;

	position: absolute;
  top: -8px;
  right: -8px;

  display: flex;
  justify-content: center;
  align-items: center;
}

.cartIconLink.sticky > .cartQuantity {
	width: 22px;
  height: 22px;
	font-size: 0.9rem;
}

.quantityWrapper{
	transform: translateY(1px);
}

.cartIconLink.sticky > .cartQuantity > .quantityWrapper {
	transform: translateY(2px);
}

@keyframes growDropdown {
  0% {
    max-height: 0;
  }
  100% {
    max-height: 500px;
  }
}

@media (max-width: 900px) {
	/* Header strandard */
  .cartIconLink > img {
		width: 45px;
	}

	.logoLink > img {
		width: 130px;
	}

	.shipyardLogo {
		font-size: 2.8rem;
	}
	/* Header sticky */
	.githubLink.sticky {
		margin: 0 1rem;
	}

	.cartIconLink.sticky {
		margin: 0 1rem;
	}

	.navbar.sticky > li {
		margin: 0 1rem;
	}
}

@media (max-width: 660px) {
	/* Header strandard */
	.logoCont {
		padding: 1rem 1rem 0 1rem;
	}

	.logoLink {
		padding-bottom: 2.5rem;
	}

	.logoLink > img {
		width: 100px;
	}

	.shipyardLink {
		position: absolute;
		top: 34px;
	}

	.shipyardLogo {
		font-size: 2.5rem;
	}

	.githubLink > img {
		width: 25px;
	}

	.cartIconLink > img {
		width: 35px;
	}

	.cartIconLink > .cartQuantity {
		width: 20px;
		height: 20px;
		font-size: 0.8rem;
	}

	.navbar > li {
		margin: 0 0.5rem;
		padding: 0 1.7rem 6px 1.7rem;
	}

	.dropdown {
		padding: 1rem 21px 1rem 21px;
		top: 37px;
	}

	.cart {
		top: 36px;
	}
	/* Header sticky */
	html {
		scroll-padding-top: 54px; 
	}

	.githubLink.sticky > img {
		width: 25px;
	}

	.cartIconLink.sticky > img {
		width: 30px;
	}

	.navbar.sticky > li {
		margin: 0;
		padding: 0 1rem;
	}

	.dropdown.sticky {
		padding: 1rem 10px 1rem 9px;
		top: 38.5px;
	}
}

@media (max-width: 500px) {
	/* Header strandard */
	.cartIconLink {
		align-self: start;
	}

	.navbar > li {
		padding: 0 1rem 6px 1rem;
	}

	.dropdown {
		padding: 1rem 13px 1rem 13px;
		font-size: 1.1rem;
		top: 37px;
	}

	.cart {
		top: 36px;
	}
	/* Header sticky */
	html {
		scroll-padding-top: 51px; 
	}

	.githubLink.sticky > img {
		width: 22px;
	}

	.cartIconLink.sticky > img {
		width: 27px;
	}

	.cartIconLink.sticky > .cartQuantity {
		width: 18px;
		height: 18px;
		font-size: 0.8rem;
	}

	.navbar.sticky > li {
		margin: 0;
		padding: 0 0.5rem;
		font-size: 1.2rem;
	}

	.dropdown.sticky {
		padding: 1rem 10px 1rem 9px;
		font-size: 1rem;
		top: 35.5px;
	}
}
