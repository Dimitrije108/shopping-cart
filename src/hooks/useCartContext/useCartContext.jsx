import { useState, useEffect, createContext, useContext } from 'react';

const CartContext = createContext({
  cart: null,
	limitExceededMsg: "", 
  addToCart: () => {},
	removeFromCart: () => {},
});
// Shopping cart functionality passed on to all components 
// via context provider
export function useShoppingCart() {
	// Retrieve cart from storage if available
	const storedCart = JSON.parse(localStorage.getItem('cart'));
	const initCart = storedCart ? storedCart : [];

	const [cart, setCart] = useState(initCart);

	useEffect(() => {
		localStorage.setItem('cart', JSON.stringify(cart));
	}, [cart]);
	// Warning message for exceeding the ship quantity limit
	const limitExceededMsg = 
		"For orders larger than 20 ships per model please contact us directly!";
	// Add an item or adjust it's quantity
	const addToCart = (item) => {
		// check if the provided item is already inside the cart
		const isDuplicate = cart.find((ship) => ship.id === item.id);
		// handle duplicate items
		if (isDuplicate) {
			const totalQuantity = isDuplicate.quantity + item.quantity;
			// handle exceeding item quantity
			if (totalQuantity > 20) {
				return false;
			};
			// adjust quantity accordingly
			const addQuantity = cart.map((ship) => {
				if (ship.id === item.id) {
					return { ...ship, quantity: ship.quantity + item.quantity };
				}
				return ship;
			});
			setCart(addQuantity);
			return true;
		} else {
			// add unique item to the cart
			setCart([...cart, item]);
			return true;
		};
	};
	// Remove item instance from the cart
	const removeFromCart = (id) => {
		const removed = cart.filter((ship) => ship.id !== id);
		setCart(removed);
	};

	return { 
		cart, 
		limitExceededMsg, 
		addToCart, 
		removeFromCart,
	};
};

export const useCartContext = () => useContext(CartContext);

export default CartContext;