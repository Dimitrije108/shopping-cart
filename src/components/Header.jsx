import { Link } from "react-router-dom";
import { useCartContext } from "../hooks/useCartContext/useCartContext";

export default function Header() {
	const { cart } = useCartContext();

  return (
		<nav>
			<ul>
				<li>
					<Link to="/">Star Wars Shipyard</Link>
				</li>
				<li>
					<Link to="starships">Starships</Link>
					<ul>
						<li>
							<Link to="starships/capital">Capital</Link>
						</li>
						<li>
							<Link to="starships/transport">Transport</Link>
						</li>
						<li>
							<Link to="starships/starfighter">Starfighter</Link>
						</li>
					</ul>
				</li>
				<li>
					<Link to="contact">Contact</Link>
				</li>
				<li>
					<Link to="shopping-cart">Cart{`(${cart.length})`}</Link>
				</li>
			</ul>
		</nav>
	)
};