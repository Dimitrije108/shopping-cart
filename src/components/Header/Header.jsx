import { useState, useEffect, useRef } from "react";
import { Link, useLocation } from "react-router-dom";
import { useCartContext } from "../../hooks/useCartContext/useCartContext";
import Navbar from "./Navbar/Navbar";
import styles from "./Header.module.css";
import layout from "../../layout.module.css";
// assets
import githubIcon from "/src/assets/icons/github-mark-white.svg";
import swLogo from "/src/assets/icons/sw-logo.png";

export default function Header() {
	const [isSticky, setIsSticky] = useState(false);
	const { pathname } = useLocation();
	const { cart } = useCartContext();

	// Check active path so custom backdrop image can be applied
  const isHomepage = pathname === "/";
  const isContact = pathname === "/contact";

	const handleSticky = (bool) => {
		setIsSticky(bool);
	};

	const header = (
		<div 
			className={`
				${styles.headerCont} 
				${isHomepage ? styles.transparentHeader : ""}
				${isContact ? styles.contactHeader : ""}
			`}
			data-testid="headerCont"
		>
			<div className={`${styles.logoCont} ${layout.layoutWrapper}`}>
				<Link 
					to="https://github.com/Dimitrije108/shopping-cart"
					className={styles.githubLink}
				>
					<img 
						src={githubIcon}
						alt="github icon" 
						title="GitHub"
					/>
				</Link>
				<h1 className={styles.logoHeading}  data-testid="headingLogo">
					<Link 
						to="/" 
						className={styles.logoLink} 
					>
						<img 
							src={swLogo}
							alt="star wars logo" 
						/>
					</Link>
					<Link 
						to="/" 
						className={styles.shipyardLink} 
						title="Shipyard"
					>
						<div className={styles.shipyardLogo}>Shipyard</div>
					</Link>
				</h1>
				<Link 
					to="shopping-cart"
					className={styles.cartIconLink}
					data-testid="cartLink"
				>
					<img 
						src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png" 
						title="Shopping Cart"
					/>
					<div className={styles.cartQuantity}>
						<div className={styles.quantityWrapper}>{`${cart.length}`}</div>
					</div>
				</Link>
			</div>
			<Navbar
				handleSticky={handleSticky}
				isSticky={isSticky}
				isMobile={false}
			/>
		</div>
	);

	const stickyHeader = (
		<div className={styles.stickyHeaderCont}>
			<div className={styles.stickyHeaderWrapper}>
				<Link 
					to="https://github.com/Dimitrije108/shopping-cart"
					className={`${styles.githubLink} ${styles.sticky}`}
				>
					<img 
						src={githubIcon}
						alt="github icon" 
						title="GitHub"
					/>
				</Link>
				<Navbar 
					handleSticky={handleSticky}
					isSticky={isSticky}
					isMobile={false}
				/>
				<Link 
					to="shopping-cart"
					className={`${styles.cartIconLink} ${styles.sticky}`}
					data-testid="cartLink"
				>
					<img 
						src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png" 
						title="Shopping Cart"
					/>
					<div className={styles.cartQuantity}>
						<div className={styles.quantityWrapper}>{`${cart.length}`}</div>
					</div>
				</Link>
			</div>
		</div>
	)

  return (
		<>{isSticky ? stickyHeader : header}</>
	)
};
