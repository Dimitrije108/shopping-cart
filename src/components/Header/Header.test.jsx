import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent  } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { useCartContext } from "../../hooks/useCartContext/useCartContext";
import Header from "./Header";

const mockCartItem = {
	id: "6429291f021f17e13fbc1db9",
	img: "https://lumiere-a.akamaihd.net/v1/images/millennium-falcon-main-tlj-a_7cf89d3a.jpeg",
	name: "Millennium Falcon",
	category: "starfighter",
	price: "100000",
	quantity: 1,
};

const mockCartItem2 = {
	id: "6429291f021f17e13fbc1e3c",
	img: "https://lumiere-a.akamaihd.net/v1/images/X-Wing-Fighter_47c7c342.jpeg",
	name: "X-wing Starfighter",
	category: "starfighter",
	price: "149999",
	quantity: 3,
};

const mockCart = [mockCartItem, mockCartItem2];

vi.mock("../../hooks/useCartContext/useCartContext", () => ({
	useCartContext: vi.fn(() => ({
		cart: mockCart,
	}))
}));

const scrollWindow = () => {
  Object.defineProperty(window, "scrollY", { value: 200, writable: true });
  fireEvent.scroll(window);
};

describe("Header component", () => {
	it("Non-sticky header snapshot", () => {
		const header = render(
			<MemoryRouter>
				<Header />
			</MemoryRouter>
		)

		expect(header).toMatchSnapshot();
	});

	it("Sticky header snapshot", () => {
		const header = render(
			<MemoryRouter>
				<Header />
			</MemoryRouter>
		)

		scrollWindow();

		expect(header).toMatchSnapshot();
	});

	describe("Non-sticky header renders correctly", () => {
		it("Homepage header loads when on homepage", () => {
			render(
				<MemoryRouter initialEntries={["/"]}>
					<Header />
				</MemoryRouter>
			)
			
			const headerContClass = screen.getByTestId("headerCont").className;

			expect(headerContClass).toMatch(/transparentHeader/);
		});

		it("Contact header loads when on contact page", () => {
			render(
				<MemoryRouter initialEntries={["/contact"]}>
					<Header />
				</MemoryRouter>
			)
			
			const headerContClass = screen.getByTestId("headerCont").className;

			expect(headerContClass).toMatch(/contactHeader/);
		});

		it("GitHub icon renders correctly and github link works", () => {
			render(
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			)

			const link = screen.getByRole("link", { name: "github icon" });
			const icon = screen.getByTitle("GitHub");

			expect(link).toBeInTheDocument();
			expect(link).toHaveAttribute("href", "https://github.com/Dimitrije108/shopping-cart");
			expect(icon).toBeInTheDocument();
			expect(icon).toHaveAttribute("src", "/src/assets/icons/github-mark-white.svg");
		});

		it("Star Wars and Shipyard logo and links render correctly", () => {
			render(
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			)

			const headingLogo = screen.getByTestId("headingLogo");
			const link1 = screen.getByRole("link", { name: "star wars logo" });
			const swLogo = screen.getByRole("img", { name: "star wars logo" });
			const link2 = screen.getByRole("link", { name: "Shipyard" });
			const shipyard = screen.getByText("Shipyard");

			expect(headingLogo).toBeInTheDocument();
			expect(link1).toBeInTheDocument();
			expect(link1).toHaveAttribute("href", "/");
			expect(swLogo).toHaveAttribute("src", "/src/assets/icons/sw-logo.png");
			expect(link2).toBeInTheDocument();
			expect(link2).toHaveAttribute("href", "/");
			expect(shipyard).toBeInTheDocument();
		});

		it("Cart icon, quantity and link render correctly", () => {
			render(
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			)

			const link = screen.getByTestId("cartLink");
			const cartIcon = screen.getByTitle("Shopping Cart");
			const cartQuantity = screen.getByText("2");

			expect(link).toBeInTheDocument();
			expect(link).toHaveAttribute("href", "/shopping-cart");
			expect(cartIcon).toBeInTheDocument();
			expect(cartIcon).toHaveAttribute("src", "https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png");
			expect(cartQuantity).toBeInTheDocument();
		});
	});

	describe("Sticky header renders correctly", () => {
		it("GitHub icon renders correctly and github link works", () => {
			render(
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			)
			
			scrollWindow();

			const link = screen.getByRole("link", { name: "github icon" });
			const icon = screen.getByTitle("GitHub");

			expect(link).toBeInTheDocument();
			expect(link).toHaveAttribute("href", "https://github.com/Dimitrije108/shopping-cart");
			expect(icon).toBeInTheDocument();
			expect(icon).toHaveAttribute("src", "/src/assets/icons/github-mark-white.svg");
		});

		it("Cart icon, quantity and link render correctly", () => {
			render(
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			)

			scrollWindow();

			const link = screen.getByTestId("cartLink");
			const cartIcon = screen.getByTitle("Shopping Cart");
			const cartQuantity = screen.getByText("2");

			expect(link).toBeInTheDocument();
			expect(link).toHaveAttribute("href", "/shopping-cart");
			expect(cartIcon).toBeInTheDocument();
			expect(cartIcon).toHaveAttribute("src", "https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png");
			expect(cartQuantity).toBeInTheDocument();
		});
	});
});
