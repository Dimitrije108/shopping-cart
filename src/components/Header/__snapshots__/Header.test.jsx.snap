// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > Non-sticky header snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
				_headerCont_63c2ea 
				_transparentHeader_63c2ea
				
			"
      >
        <div
          class="_logoCont_63c2ea _layoutWrapper_aadc39"
        >
          <a
            class="_githubLink_63c2ea"
            href="https://github.com/Dimitrije108/shopping-cart"
          >
            <img
              alt="github icon"
              src="/src/assets/github-mark-white.svg"
              title="GitHub"
              width="30"
            />
          </a>
          <h1
            class="_logoHeading_63c2ea"
            data-testid="headingLogo"
          >
            <a
              class="_logoLink_63c2ea"
              data-discover="true"
              href="/"
            >
              <img
                alt="star wars logo"
                src="/src/assets/sw-logo.png"
                width="150"
              />
            </a>
            <a
              class="_shipyardLink_63c2ea"
              data-discover="true"
              href="/"
              title="Shipyard"
            >
              <div
                class="_shipyardLogo_63c2ea"
              >
                Shipyard
              </div>
            </a>
          </h1>
          <a
            class="_cartIconLink_63c2ea"
            data-discover="true"
            data-testid="cartLink"
            href="/shopping-cart"
          >
            <img
              height="55"
              src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png"
              title="Shopping Cart"
              width="55"
            />
            <div
              class="_cartQuantity_63c2ea"
            >
              <div
                class="_quantityWrapper_63c2ea"
              >
                2
              </div>
            </div>
          </a>
        </div>
        <nav
          class="_navbarCont_63c2ea"
        >
          <div />
          <ul
            class="_navbar_63c2ea"
          >
            <li
              class="
							_homeNav_63c2ea 
							_active_63c2ea 
						"
              data-testid="home"
            >
              <a
                data-discover="true"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="
							_starshipsNav_63c2ea 
							 
						"
              data-testid="starships"
            >
              <a
                data-discover="true"
                href="/starships"
              >
                Starships
              </a>
              <ul
                class="
								_dropdown_63c2ea 
								
							"
                data-testid="starshipsDropdown"
              >
                <li>
                  <a
                    data-discover="true"
                    href="/starships/capital"
                  >
                    Capital
                  </a>
                </li>
                <li>
                  <a
                    data-discover="true"
                    href="/starships/transport"
                  >
                    Transport
                  </a>
                </li>
                <li>
                  <a
                    data-discover="true"
                    href="/starships/starfighter"
                  >
                    Starfighter
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="
							_contactNav_63c2ea 
							 
						"
              data-testid="contact"
            >
              <a
                data-discover="true"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
				_headerCont_63c2ea 
				_transparentHeader_63c2ea
				
			"
    >
      <div
        class="_logoCont_63c2ea _layoutWrapper_aadc39"
      >
        <a
          class="_githubLink_63c2ea"
          href="https://github.com/Dimitrije108/shopping-cart"
        >
          <img
            alt="github icon"
            src="/src/assets/github-mark-white.svg"
            title="GitHub"
            width="30"
          />
        </a>
        <h1
          class="_logoHeading_63c2ea"
          data-testid="headingLogo"
        >
          <a
            class="_logoLink_63c2ea"
            data-discover="true"
            href="/"
          >
            <img
              alt="star wars logo"
              src="/src/assets/sw-logo.png"
              width="150"
            />
          </a>
          <a
            class="_shipyardLink_63c2ea"
            data-discover="true"
            href="/"
            title="Shipyard"
          >
            <div
              class="_shipyardLogo_63c2ea"
            >
              Shipyard
            </div>
          </a>
        </h1>
        <a
          class="_cartIconLink_63c2ea"
          data-discover="true"
          data-testid="cartLink"
          href="/shopping-cart"
        >
          <img
            height="55"
            src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png"
            title="Shopping Cart"
            width="55"
          />
          <div
            class="_cartQuantity_63c2ea"
          >
            <div
              class="_quantityWrapper_63c2ea"
            >
              2
            </div>
          </div>
        </a>
      </div>
      <nav
        class="_navbarCont_63c2ea"
      >
        <div />
        <ul
          class="_navbar_63c2ea"
        >
          <li
            class="
							_homeNav_63c2ea 
							_active_63c2ea 
						"
            data-testid="home"
          >
            <a
              data-discover="true"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="
							_starshipsNav_63c2ea 
							 
						"
            data-testid="starships"
          >
            <a
              data-discover="true"
              href="/starships"
            >
              Starships
            </a>
            <ul
              class="
								_dropdown_63c2ea 
								
							"
              data-testid="starshipsDropdown"
            >
              <li>
                <a
                  data-discover="true"
                  href="/starships/capital"
                >
                  Capital
                </a>
              </li>
              <li>
                <a
                  data-discover="true"
                  href="/starships/transport"
                >
                  Transport
                </a>
              </li>
              <li>
                <a
                  data-discover="true"
                  href="/starships/starfighter"
                >
                  Starfighter
                </a>
              </li>
            </ul>
          </li>
          <li
            class="
							_contactNav_63c2ea 
							 
						"
            data-testid="contact"
          >
            <a
              data-discover="true"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header component > Sticky header snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
				_headerCont_63c2ea 
				_transparentHeader_63c2ea
				
			"
      >
        <nav
          class="_navbarCont_63c2ea _sticky_63c2ea"
          data-testid="stickyNavbar"
        >
          <a
            class="_githubLink_63c2ea _sticky_63c2ea"
            href="https://github.com/Dimitrije108/shopping-cart"
          >
            <img
              alt="github icon"
              src="/src/assets/github-mark-white.svg"
              title="GitHub"
              width="28"
            />
          </a>
          <ul
            class="_navbar_63c2ea"
          >
            <li
              class="
						_homeNav_63c2ea 
						_active_63c2ea 
					"
              data-testid="home"
            >
              <a
                data-discover="true"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="
						_starshipsNav_63c2ea 
						 
					"
              data-testid="starships"
            >
              <a
                data-discover="true"
                href="/starships"
              >
                Starships
              </a>
              <ul
                class="
							_dropdown_63c2ea 
							_sticky_63c2ea 
							
							"
                data-testid="starshipsDropdown"
              >
                <li>
                  <a
                    data-discover="true"
                    href="/starships/capital"
                  >
                    Capital
                  </a>
                </li>
                <li>
                  <a
                    data-discover="true"
                    href="/starships/transport"
                  >
                    Transport
                  </a>
                </li>
                <li>
                  <a
                    data-discover="true"
                    href="/starships/starfighter"
                  >
                    Starfighter
                  </a>
                </li>
              </ul>
            </li>
            <li
              class="
						_contactNav_63c2ea 
						 
					"
              data-testid="contact"
            >
              <a
                data-discover="true"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
          <a
            class="_cartIconLink_63c2ea _sticky_63c2ea"
            data-discover="true"
            data-testid="cartLink"
            href="/shopping-cart"
          >
            <img
              src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png"
              title="Shopping Cart"
              width="35"
            />
            <div
              class="_cartQuantity_63c2ea"
            >
              <div
                class="_quantityWrapper_63c2ea"
              >
                2
              </div>
            </div>
          </a>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
				_headerCont_63c2ea 
				_transparentHeader_63c2ea
				
			"
    >
      <nav
        class="_navbarCont_63c2ea _sticky_63c2ea"
        data-testid="stickyNavbar"
      >
        <a
          class="_githubLink_63c2ea _sticky_63c2ea"
          href="https://github.com/Dimitrije108/shopping-cart"
        >
          <img
            alt="github icon"
            src="/src/assets/github-mark-white.svg"
            title="GitHub"
            width="28"
          />
        </a>
        <ul
          class="_navbar_63c2ea"
        >
          <li
            class="
						_homeNav_63c2ea 
						_active_63c2ea 
					"
            data-testid="home"
          >
            <a
              data-discover="true"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="
						_starshipsNav_63c2ea 
						 
					"
            data-testid="starships"
          >
            <a
              data-discover="true"
              href="/starships"
            >
              Starships
            </a>
            <ul
              class="
							_dropdown_63c2ea 
							_sticky_63c2ea 
							
							"
              data-testid="starshipsDropdown"
            >
              <li>
                <a
                  data-discover="true"
                  href="/starships/capital"
                >
                  Capital
                </a>
              </li>
              <li>
                <a
                  data-discover="true"
                  href="/starships/transport"
                >
                  Transport
                </a>
              </li>
              <li>
                <a
                  data-discover="true"
                  href="/starships/starfighter"
                >
                  Starfighter
                </a>
              </li>
            </ul>
          </li>
          <li
            class="
						_contactNav_63c2ea 
						 
					"
            data-testid="contact"
          >
            <a
              data-discover="true"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
        <a
          class="_cartIconLink_63c2ea _sticky_63c2ea"
          data-discover="true"
          data-testid="cartLink"
          href="/shopping-cart"
        >
          <img
            src="https://icons.iconarchive.com/icons/jonathan-rey/star-wars-vehicles/128/Death-Star-2nd-icon.png"
            title="Shopping Cart"
            width="35"
          />
          <div
            class="_cartQuantity_63c2ea"
          >
            <div
              class="_quantityWrapper_63c2ea"
            >
              2
            </div>
          </div>
        </a>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
